mypy/checker.py

```diff
@@ -1716,7 +1749,7 @@ def split_around_star(self, items: List[T], star_index: int,
         returns in: ([1,2], [3,4,5], [6,7])
         """
         nr_right_of_star = length - star_index - 1
-        right_index = nr_right_of_star if -nr_right_of_star != 0 else len(items)
+        right_index = -nr_right_of_star if nr_right_of_star != 0 else len(items)
```

Comment:
This bug was exposed by one of the tests. It's quite surprising that it never appeared before, since the old logic took _first_ ``nr_right_of_star`` types instead of the same number counting from the end of ``items``.